cpp/cmake_modules/ThirdpartyToolchain.cmake

```
@@ -5039,6 +5054,154 @@ if(ARROW_S3)
+  list(GET OPENSSL_VERSION_ELEMENTS_LIST 0 OPENSSL_VERSION_MAJOR)
+  if(OPENSSL_VERSION_MAJOR EQUAL 1)
+    list(APPEND
+    AZURESDK_COMMON_CMAKE_ARGS
+    -DVCPKG_OVERLAY_PORTS=${CMAKE_CURRENT_BINARY_DIR}/azuresdk_ep-prefix/src/azuresdk_ep/vcpkg-custom-ports)
+  elseif(OPENSSL_VERSION_MAJOR EQUAL 3)
+  else()
+    message(FATAL_ERROR "Unknown OpenSSL version: ${OPENSSL_VERSION}")
+  endif()
+
+  file(MAKE_DIRECTORY ${AZURESDK_INCLUDE_DIR})
+  set(AZURE_CORE_STATIC_LIBRARY
+      "${AZURESDK_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}azure-core${CMAKE_STATIC_LIBRARY_SUFFIX}"
+  )
+  set(AZURE_IDENTITY_STATIC_LIBRARY
+      "${AZURESDK_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}azure-identity${CMAKE_STATIC_LIBRARY_SUFFIX}"
+  )
+  set(AZURE_STORAGE_BLOBS_STATIC_LIBRARY
+      "${AZURESDK_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}azure-storage-blobs${CMAKE_STATIC_LIBRARY_SUFFIX}"
+  )
+  set(AZURE_STORAGE_COMMON_STATIC_LIBRARY
+      "${AZURESDK_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}azure-storage-common${CMAKE_STATIC_LIBRARY_SUFFIX}"
+  )
+  set(AZURE_STORAGE_FILES_DATALAKE_STATIC_LIBRARY
+      "${AZURESDK_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}azure-storage-files-datalake${CMAKE_STATIC_LIBRARY_SUFFIX}"
+  )
+  externalproject_add(azuresdk_ep
+                      ${EP_COMMON_OPTIONS}
+                      INSTALL_DIR ${AZURESDK_PREFIX}
+                      URL ${ARROW_AZURESDK_URL}
+                      URL_HASH "SHA256=${ARROW_AZURE_SDK_BUILD_SHA256_CHECKSUM}"
+                      CMAKE_ARGS ${AZURESDK_COMMON_CMAKE_ARGS}
+                      BUILD_BYPRODUCTS ${AZURE_CORE_STATIC_LIBRARY}
+                                       ${AZURE_IDENTITY_STATIC_LIBRARY}
+                                       ${AZURE_STORAGE_BLOBS_STATIC_LIBRARY}
+                                       ${AZURE_STORAGE_COMMON_STATIC_LIBRARY}
+                                       ${AZURE_STORAGE_FILES_DATALAKE_STATIC_LIBRARY}
+                      DEPENDS azure_sdk_dependencies)
+  add_library(Azure::azure-core STATIC IMPORTED)
+  set_target_properties(Azure::azure-core
+                        PROPERTIES IMPORTED_LOCATION ${AZURE_CORE_STATIC_LIBRARY}
+                                   INTERFACE_INCLUDE_DIRECTORIES
+                                   ${AZURESDK_INCLUDE_DIR})
+  set_property(TARGET Azure::azure-core
+               PROPERTY INTERFACE_LINK_LIBRARIES CURL::libcurl OpenSSL::Crypto OpenSSL::SSL)
+  add_dependencies(Azure::azure-core azuresdk_ep)
+
+  add_library(Azure::azure-identity STATIC IMPORTED)
+  set_target_properties(Azure::azure-identity 
+                        PROPERTIES IMPORTED_LOCATION ${AZURE_IDENTITY_STATIC_LIBRARY}
+                                   INTERFACE_INCLUDE_DIRECTORIES
+                                   ${AZURESDK_INCLUDE_DIR})
+  set_property(TARGET Azure::azure-identity
+               PROPERTY INTERFACE_LINK_LIBRARIES OpenSSL::Crypto Azure::azure-core)
```

`Azure::azure-core` already depends on `OpenSSL::Crypto`.

```suggestion
               PROPERTY INTERFACE_LINK_LIBRARIES Azure::azure-core)
```

---

https://github.com/Azure/azure-sdk-for-cpp/blob/c2b2d565794c53992d1523ddb4df77a3292df628/sdk/identity/azure-identity/CMakeLists.txt#L103 https://github.com/Azure/azure-sdk-for-cpp/blob/c2b2d565794c53992d1523ddb4df77a3292df628/sdk/identity/azure-identity/src/client_certificate_credential.cpp#L23

---

Thanks.
It seems that azure-core doesn't depend on `OpenSSL::Crypto`: https://github.com/Azure/azure-sdk-for-cpp/blob/c2b2d565794c53992d1523ddb4df77a3292df628/sdk/core/azure-core/CMakeLists.txt#L178
How about the followings?
* `Azure::azure-core` depends on only `CURL::libcurl` and `OpenSSL::SSL`
* `Azure::azure-identity` depends on `Azure::azure-core` and `OpenSSL::Crypto`

---

Probably I should have tried harder to get that working before but its working now :slightly_smiling_face:

---

Probably I should have tried harder to get that working before but its working now :slightly_smiling_face:

---

Done. I also double checked the relevant `CMakeLists.txt` for all the other components of the Azure SDK and explicitly declared exactly the dependencies required according to each `CMakeLists.txt`.

---

@kou are you happy with applying this same policy to all the other components. If so I'll mark the repetitions of this comment as resolved.